### ISSUE TITLE
Inconsistent/buggy focus for "Search this area" button

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#17665]
## What happened?
During his validation for [#6911], Josh observed the inconsistent focus management between zoom in/out and "search this area". Engineers have addressed the announcement of the button and change in context in a way that is appropriate for users who may need to adjust the zoom level by interacting with the controls more than once. 

one additional tweak - we need to provide Voiceover instructions for how the button can be reached (ie: _you can reach the "Search this area" button by tabbing..._)

## Desired behavior
Veterans using voiceover should how to access the button when it receives focus 

## Acceptance Criteria
- [ ] A Veteran using Safari/Voiceover can use the space or enter keys to move focus to the "search this area" button.

**Comment by mmiddaugh** [comment]
Please refer to [recording with Josh](https://drive.google.com/file/d/1drsmPly33CQHGzVsxrvY_0y9OmMiHrZR/view?usp=sharing)

**Comment by mmiddaugh** [comment]
Needs design input for UX - will confirm with Leyda
Per Josh, we are handling focus appropriately _if the intended behavior is to give the Veteran control_

**Comment by mmiddaugh** [comment]
@brangi I think we've introduced some odd behaviors with this change. Please see [this Loom](https://www.loom.com/share/bae53b16a6af4b238790d0ea2ec1d57f). 

Search parameters: Polson, Montana + Community care providers + Optometrist

After a successful search, I moved the map and initiated a new search using the "Search this area button". The map **correctly** displayed the area I wished to search, appeared to do a new search but then returned the same results which were not within that area. When I moved the map farther away, it correctly returned no results. 

I was able to replicate the experience by moving back to a larger area to get a new, larger set of results. When I zoomed into a closer area of the map, the locator appeared to search the new area but again returned the previous set of results  (which were not contained within the search area).

**Comment by mmiddaugh** [comment]
@joshkimux this is ready for your review when you are able

**Comment by joshkimux** [comment]
@mmiddaugh This is working great! The search this area of the map is available in the tab order, and focus stays on the zoom in and zoom out buttons. When the search this area of the map button appears it is also announced. 🥳 

One issue did stand out to me, which I hope can be a relatively quick fix. Currently, upon reloading the page (after allowing for the browser to use my location), the "search this area of the map" will still steal focus away from the user. This also occurs when using the normal search button as well.

Would it be possible for us to not shift focus to the "search this area of the map" on page load and on firing the normal search button?

https://user-images.githubusercontent.com/14154792/110360603-eb440b00-800c-11eb-9379-c50a380d7b4e.mov

**Comment by mpelzsherman** [comment]
We need a ruling on the UX for messaging the user if geolocation fails due to lack of permission.
Attn @lildziner and @joshkimux

**Comment by mmiddaugh** [comment]
The remainder of this work will be captured in #21256 and #21958
